src/ParamConverter/HashidsParamConverter.php

```diff
@@ -39,53 +40,79 @@ public function supports(ParamConverter $configuration): bool
         return true;
     }
 
-    private function setHashid(Request $request, ParamConverter $configuration): void
+    private function decodeHashidOnRoute(Request $request): bool
+    {
+        foreach (['hashid', 'id'] as $item) {
+            $hashids = $this->hashids->decode($request->attributes->get($item));
+            if ($this->hasHashidDecoded($hashids)) {
+                $hashid = current($hashids);
+
+                $request->attributes->remove($item);
+                $request->attributes->set('id', $hashid);
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function decodeArgumentsController(Request $request, ParamConverter $configuration): void
     {
-        $hashids = $this->hashids->decode(
-            $this->getIdentifier(
-                $request,
-                array_replace(['hashid' => null], $configuration->getOptions()),
-                (string) $configuration->getName()
-            )
-        );
+        $name = $configuration->getName();
+        $hash = $this->getHash($request, $name);
+
+        if ($this->isSkippable($hash)) {
+            return;
+        }
+
+        $hashids = $this->hashids->decode($hash);
 
         if ($this->hasHashidDecoded($hashids)) {
-            $request->attributes->set($configuration->getName(), current($hashids));
+            $request->attributes->set($name, current($hashids));
+        }
+
+        if (!$this->autoConvert && !$this->hasHashidDecoded($hashids)) {
+            throw new \LogicException(sprintf('Unable to decode parameter "%s".', $name));
         }
     }
 
-    private function getIdentifier(Request $request, $options, string $name): string
+    private function getHash(Request $request, string $name): ?string
     {
-        if ($options['hashid'] && !is_array($options['hashid'])) {
-            $name = $options['hashid'];
+        if (empty($name)) {
+            return null;
         }
 
-        if ($request->attributes->has($name)) {
-            return (string) $request->attributes->get($name);
+        $attributes = array_map(function ($name) {
+            return str_replace('_hash_', '', $name);
+        }, $request->attributes->keys());
+
+        $hash = null;
+
+        if (in_array($name, $attributes, true)) {
```

Comment:
instead of this, you could directly look for the attribute with `$request->attributes->has('_hash'.$name)` and `$request->attributes->has($name)`.

It is easier, and will be more performant than `in_array()`

Reply:
Done